lang/src/org/partiql/lang/ast/passes/AstRewriterBase.kt

```diff
@@ -31,8 +32,9 @@ interface AstRewriter {
 @Deprecated("New rewriters should implement PIG's PartiqlAst.VisitorTransform instead")
 open class AstRewriterBase : AstRewriter {
 
-    override fun rewriteExprNode(node: ExprNode): ExprNode =
-        when (node) {
+    override fun rewriteExprNode(node: ExprNode): ExprNode {
+        checkThreadInterrupted()
```

Comment:
I think would have to see what that would look like in practice to offer any sort of strong opinion on the matter.  

In any case, I think it would require some modification of PIG, which is of course beyond the current scope.